babi/screen.py

```diff
@@ -518,8 +517,15 @@ def save(self) -> PromptResult | None:
             ) as f:
                 f.write(contents)
         except OSError as e:
-            self.status.update(f'cannot save file: {e}')
-            return PromptResult.CANCELLED
+            if e.errno == 2:
+                os.makedirs(os.path.dirname(self.file.filename), exist_ok=True)
```

Comment:
ah ok, yeah I mean that this should be the only line added, and above the `with open` and nothing else needs to change (because `makedirs` + `exist_ok=True` is idempotent)

Reply:
oh, i think i just slowly get what you mean...